added hiscore support to invaders

shaunquick opened the pull request:
added hiscore support - re-worked DrvFrame to ensure hiscores work correctly.

dinkc64 commented:
Hi shaunquick,
After looking at your PR, I noticed something was a little off in the invaders driver,
"if (watchdog >= 180) {"
replace 180 with 255, and now the only changes that need to be made to the driver are:
HiscoreReset() in DrvDoReset() and the BDF_HISCORE_SUPPORTED flags for the games.  Could you please submit it with just those changes?

best regards,
- dink

shaunquick commented:
Hi dink,
    After sorting out my finger trouble and reversion of code, I have changed 
"if (watchdog >= 180) { to "if (watchdog >= 255) {"
 and added hi scores (and reverted back DrvDoReset to have the parameter) 
Apologies for the multiple commits - if need be I can delete my fork and restart. Please let me know.

Regards

barbudreadmon commented:
>Apologies for the multiple commits - if need be I can delete my fork and restart. Please let me know.

That's not gonna be a problem, we can squash them together when merging :)
